hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminDefaultMonitor.java

```diff
@@ -270,12 +274,34 @@ private void check() {
         // an invalid state.
         LOG.warn("DatanodeAdminMonitor caught exception when processing node "
             + "{}.", dn, e);
-        pendingNodes.add(dn);
-        toRemove.add(dn);
+        toRequeue.add(dn);
```

Comment:
I suppose reverting this change seems more simple because it can remove `toRequeue` variable.
```suggestion
        getPendingNodes.add(dn);
        toRemove.add(dn);
```

Reply:
I have modified the code to remove the toRequeue variable